File: production/shared/src/main/scala/kpn/shared/route/RouteChangeInfo.scala
package kpn.shared.route

import kpn.shared.Bounds
import kpn.shared.changes.ChangeSetInfo
import kpn.shared.changes.details.ChangeKey
import kpn.shared.data.MetaData
import kpn.shared.data.raw.RawNode
import kpn.shared.diff.WayInfo
import kpn.shared.diff.WayUpdate
import kpn.shared.diff.route.RouteDiff

case class RouteChangeInfo(
  id: Long,
  version: Int,
  changeKey: ChangeKey,
  comment: Option[String],
  before: Option[MetaData],
  after: Option[MetaData],

  removedWays: Seq[WayInfo],
  addedWays: Seq[WayInfo],
  updatedWays: Seq[WayUpdate],

  diffs: RouteDiff,
  nodes: Seq[RawNode],
  changeSetInfo: Option[ChangeSetInfo] = None,
  addedNodes: Seq[Long] = Seq.empty,
  deletedNodes: Seq[Long] = Seq.empty,
  commonNodes: Seq[Long] = Seq.empty,
  addedWayIds: Seq[Long] = Seq.empty,
  deletedWayIds: Seq[Long] = Seq.empty,
  commonWayIds: Seq[Long] = Seq.empty,
  geometryDiff: Option[GeometryDiff] = None,
  bounds: Bounds = Bounds(),
  happy: Boolean = false,
  investigate: Boolean = false
)

File: backend/api/src/main/java/kpn/api/common/route/RouteChangeInfo.java
package kpn.api.common.route;

import kpn.api.common.ChangeType;
import kpn.api.common.changes.ChangeSetInfo;
import kpn.api.common.changes.details.ChangeKey;
import kpn.api.common.data.MetaData;
import kpn.api.common.diff.WayDiffsInfo;
import kpn.api.common.diff.route.RouteDiff;
import kpn.api.common.route.GeometryDiff;
import kpn.api.common.route.RouteNode;
import kpn.api.common.route.RouteNodeChange;

import java.util.Optional;
import com.google.common.collect.ImmutableList;

public record RouteChangeInfo(
  Long rowIndex,
  Long id,
  Long version,
  ChangeKey changeKey,
  ChangeType changeType,
  Optional<String> comment,
  Optional<MetaData> before,
  Optional<MetaData> after,
  Optional<RouteDiff> diffs,
  ImmutableList<RouteNode> nodes,
  ImmutableList<RouteNodeChange> nodeChanges,
  Optional<ChangeSetInfo> changeSetInfo,
  Optional<WayDiffsInfo> wayDiffs,
  Optional<GeometryDiff> geometryDiff,
  Boolean happy,
  Boolean investigate
) {
}

/*
package kpn.api.common.route

import kpn.api.common.ChangeType
import kpn.api.common.changes.ChangeSetInfo
import kpn.api.common.changes.details.ChangeKey
import kpn.api.common.data.MetaData
import kpn.api.common.diff.WayDiffsInfo
import kpn.api.common.diff.route.RouteDiff

case class RouteChangeInfo(
  rowIndex: Long,
  id: Long,
  version: Long,
  changeKey: ChangeKey,
  changeType: ChangeType,
  comment: Option[String],
  before: Option[MetaData],
  after: Option[MetaData],
  diffs: Option[RouteDiff],
  nodes: Seq[RouteNode],
  nodeChanges: Seq[RouteNodeChange],
  changeSetInfo: Option[ChangeSetInfo],
  wayDiffs: Option[WayDiffsInfo],
  geometryDiff: Option[GeometryDiff],
  happy: Boolean,
  investigate: Boolean
)

*/
